beacon-chain/core/epoch/epoch_operations.go

```diff
@@ -0,0 +1,338 @@
+//    Let total_balance(shard_committee) =
+//    sum([get_effective_balance(state, i) for i in shard_committee.committee])
+func TotalBalance(
+	state *pb.BeaconState,
+	shardCommittee *pb.ShardAndCommittee) uint64 {
+
+	var totalBalance uint64
+	for _, index := range shardCommittee.Committee {
+		totalBalance += validators.EffectiveBalance(state, index)
+	}
+
+	return totalBalance
+}
+
+// InclusionSlot returns the slot number of when the validator's
+// attestation gets included in the beacon chain.
+//
+// Spec pseudocode definition:
+//    Let inclusion_slot(state, index) =
+//    a.slot_included for the attestation a where index is in
+//    get_attestation_participants(state, a.data, a.participation_bitfield)
+func InclusionSlot(state *pb.BeaconState, validatorIndex uint32) (uint64, error) {
+
+	for _, attestation := range state.LatestAttestations {
+		participatedValidators, err := validators.AttestationParticipants(state, attestation.Data, attestation.ParticipationBitfield)
+		if err != nil {
+			return 0, fmt.Errorf("could not get attestation participants: %v", err)
+		}
+
+		for _, index := range participatedValidators {
+			if index == validatorIndex {
+				return attestation.SlotIncluded, nil
+			}
+		}
+	}
+	return 0, fmt.Errorf("could not find inclusion slot for validator index %d", validatorIndex)
+}
+
+// InclusionDistance returns the difference in slot number of when attestation
+// gets submitted and when it gets included.
+//
+// Spec pseudocode definition:
+//    Let inclusion_distance(state, index) =
+//    a.slot_included - a.data.slot where a is the above attestation same as
+//    inclusion_slot
+func InclusionDistance(state *pb.BeaconState, validatorIndex uint32) (uint64, error) {
+
+	for _, attestation := range state.LatestAttestations {
+		participatedValidators, err := validators.AttestationParticipants(state, attestation.Data, attestation.ParticipationBitfield)
+		if err != nil {
+			return 0, fmt.Errorf("could not get attestation participants: %v", err)
+		}
+
+		for _, index := range participatedValidators {
+			if index == validatorIndex {
+				return attestation.SlotIncluded - attestation.Data.Slot, nil
+			}
+		}
+	}
+	return 0, fmt.Errorf("could not find inclusion distance for validator index %d", validatorIndex)
+}
+
+// AdjustForInclusionDistance returns the calculated reward based on
+// how long it took for attestation to get included. The longer, the lower
+// the the reward.
+//
+// Spec pseudocode definition:
+//    def adjust_for_inclusion_distance(magnitude: int, distance: int) -> int:
+//    """
+//    Adjusts the reward of an attestation based on how long it took to get included
+//    (the longer, the lower the reward). Returns a value between ``0`` and ``magnitude``.
+//    ""
+//    return magnitude // 2 + (magnitude // 2) * MIN_ATTESTATION_INCLUSION_DELAY // distance
+func AdjustForInclusionDistance(magniture uint64, distance uint64) uint64 {
+	return magniture/2 + (magniture/2)*
```

Comment:
Nitpick, You could factor it out and write it as:
`(magniture/2) * (1 + params.BeaconConfig().MinAttestationInclusionDelay/distance )`

Reply:
I was thinking that but the current format gives you more readability. You can easily tell the magnitude increases the reward factor while distance decreases the reward factor `MIN_ATTESTATION_INCLUSION_DELAY // 2 * distance`